src/main/java/frc/robot/subsystems/auto/BargeAlign.java

```diff
@@ -0,0 +1,113 @@
+  private PIDController pidX = new PIDController(4, 0, 0);
+  private PIDController pidRotate = new PIDController(8, 0, 0);
+
+  private CommandSwerveDrivetrain drive;
+  private static final double fieldLength = 17.548; // Welded field
+  private static final double blueBlacklineX = 7.09;
+  private static final double redBlacklineX = fieldLength - blueBlacklineX;
+  private static final double blueBargeLineX = 7.67;
+  private static final double redBargeLineX = fieldLength - blueBargeLineX;
+
+  private final SwerveRequest.FieldCentric blackLineDriveRequest =
+      new SwerveRequest.FieldCentric() // Add a 10% deadband
+          .withDriveRequestType(DriveRequestType.OpenLoopVoltage)
+          .withForwardPerspective(ForwardPerspectiveValue.BlueAlliance);
+
+  public static boolean atScoringXPosition(CommandSwerveDrivetrain drivebasesubsystem) {
+    double robotX = drivebasesubsystem.getState().Pose.getX();
+    return blueBargeLineX < robotX && robotX < redBargeLineX;
+  }
+
+  public static Command driveToBlackLine(CommandSwerveDrivetrain drivebaseSubsystem) {
+    return new BargeAlign(drivebaseSubsystem).withName("Drive to Black Line");
+  }
+
+  private static final SwerveRequest.FieldCentric bargeDriveRequest =
+      new SwerveRequest.FieldCentric()
+          .withDeadband(0.0001)
+          .withRotationalDeadband(0.0001)
+          .withDriveRequestType(DriveRequestType.OpenLoopVoltage)
+          .withForwardPerspective(ForwardPerspectiveValue.BlueAlliance);
+
+  private static final double xBargeDriveSpeed = 0.5;
+
+  public static Command driveToBarge(CommandSwerveDrivetrain drivebaseSubsystem) {
+    return drivebaseSubsystem
+        .applyRequest(
+            () -> {
+              boolean onRedSide = drivebaseSubsystem.getState().Pose.getX() > fieldLength / 2;
+              return bargeDriveRequest
+                  .withVelocityX(onRedSide ? -xBargeDriveSpeed : xBargeDriveSpeed)
+                  .withVelocityY(0)
+                  .withRotationalRate(0);
+            })
+        .finallyDo(
+            () ->
+                drivebaseSubsystem.setControl(
+                    bargeDriveRequest.withVelocityX(0).withVelocityY(0).withRotationalRate(0)))
+        .withName("Drive to barge");
+  }
+
+  private BargeAlign(CommandSwerveDrivetrain drive) {
+    this.drive = drive;
+    pidRotate.enableContinuousInput(-Math.PI, Math.PI);
+    addRequirements(drive);
+    setName("drive to black line");
+  }
+
+  @Override
+  public void initialize() {
+    boolean onRedSide = drive.getState().Pose.getX() > fieldLength / 2;
+    double targetX = onRedSide ? redBlacklineX : blueBlacklineX;
+    Rotation2d targetAngle = onRedSide ? Rotation2d.k180deg : Rotation2d.kZero;
+    pidX.setSetpoint(targetX);
+    pidRotate.setSetpoint(targetAngle.getRadians());
+  }
+
+  @Override
+  public void execute() {
+    Pose2d currentPose = drive.getState().Pose;
+    // Calculate the power for X direction and clamp it between -1 and 1
```

Comment:
? Comment says 'clamp it between -1 and 1' but the code appears to clamp between -2 and 2